src/PreferenceDB.java

```diff
@@ -0,0 +1,56 @@
+import Preferences.Preferences;
+import Preferences.PreferencesBuilder;
+import Preferences.PreferencesBuilderImpl;
+import channel.ChannelType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+public class PreferenceDB {
+    // stores user id and its corresponding preferences
+    private final HashMap<Integer, Preferences> preferencesData;
```

Comment:
make this static so that only one copy should create. This is more like config which store every one preferences.